src/ff/prime_field.rs

```diff
@@ -0,0 +1,307 @@
+use super::{field::BinaryField, Field};
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt::Debug,
+    ops::{
+        Add, AddAssign, BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Mul,
+        MulAssign, Neg, Not, Sub, SubAssign,
+    },
+};
+
+macro_rules! field_impl {
+    ( $( $field:ty ),* ) => { $(
+        impl Add for $field {
```

Comment:
To make this macro usable more generally (that is, without having to add `use` statements for all of the types the macro refers to), you should make this:

```suggestion
        impl std::ops::Add for $field {
```
Basically, use a fully-qualified name for everything where possible.